ledger/block/src/verify.rs

```diff
@@ -563,10 +563,15 @@ impl<N: Network> Block<N> {
             // Process the transmission ID.
             match transmission_id {
                 TransmissionID::Ratification => {}
-                TransmissionID::Solution(solution_id) => {
+                TransmissionID::Solution(solution_id, checksum) => {
                     match solutions.peek() {
                         // Check the next solution matches the expected solution ID.
-                        Some((_, solution)) if solution.id() == *solution_id => {
+                        Some((_, solution))
+                            if solution.id() == *solution_id
+                                && Data::<Solution<N>>::Buffer(solution.to_bytes_le()?.into())
```

Comment:
we might want to introduce (probably in `snarkvm-utilities`) a free function (with a bound on `Network`) that converts a `&[u8]` to a checksum in order to avoid having to use a `Data` wrapper here and in a few other spots; that function would then be reused internally in `Data::to_checksum`.

Reply:
We could also consider implementing `Eq` between the solution and the `Data` wrapper type directly so her it could be simplified to `solution == *checksum`.